### PR TITLE
Move meta data to tenant space

### PR DESCRIPTION
**Pull request by senthalan**
### Proposed changes in this pull request
Move the following caches to tenant space
- OAuthScopeBindingCache
- OAuthScopeCache
- ClaimCache
- ClaimMetaDataCache
- OIDCScopeClaimCache

Related to https://github.com/wso2/product-is/issues/11690

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/OAuthScopeBindingCache.java [hunk 1]
```diff
@@ -53,10 +53,11 @@ public static OAuthScopeBindingCache getInstance() {
      *
      * @param key   Key which cache entry is indexed.
      * @param entry Actual object where cache entry is placed.
+     * @param tenantId Tenant where the cache is maintained.
      */
-    public void addToCache(OAuthScopeBindingCacheKey key, Scope[] entry) {
+    public void addToCache(OAuthScopeBindingCacheKey key, Scope[] entry, int tenantId) {
```
[the remark this answers is not in the document]
Comment (senthalan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/OAuthScopeBindingCache.java:58]: yeah. we are moving all the caches to the tenant space base by introducing a new base cache. we need to pass the correct tenant domain for each cache operatation.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/CacheEntry.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
  * cache must be encapsulated with a class deriving from this class.
  */
 @SuppressFBWarnings("NM_SAME_SIMPLE_NAME_AS_SUPERCLASS")
-public abstract class CacheEntry extends org.wso2.carbon.identity.application.common.cache.CacheEntry {
+public abstract class CacheEntry extends org.wso2.carbon.identity.core.cache.CacheEntry {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/CacheEntry.java:28]: Shouldn't we change serialVersionUID?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/cache/OIDCScopeClaimCacheEntry.java [hunk 3]
```diff
@@ -18,7 +18,7 @@
 
 package org.wso2.carbon.identity.openidconnect.cache;
 
-import org.wso2.carbon.identity.application.common.cache.CacheEntry;
+import org.wso2.carbon.identity.core.cache.CacheEntry;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/cache/OIDCScopeClaimCacheEntry.java:21]: Shouldn't we fix the usage of this cache to get the tenant id in the getters?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/util/ClaimMetaDataCache.java [hunk 4]
```diff
@@ -28,10 +32,15 @@
     private static final String CLAIM_META_DATA_CACHE_NAME = "ClaimMetaDataCache";
 
     private static ClaimMetaDataCache instance = new ClaimMetaDataCache();
+    private static final List<AbstractCacheListener<ClaimMetaDataCacheKey, ClaimMetaDataCacheEntry>>
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/util/ClaimMetaDataCache.java:35]: Do we need these listeners now?
Remember they were used when we had distributed caches. If they are not used can we get rid of them now.

**Review by mefarazath — APPROVED** [review]

**Pull request merged**